lib/shared/addon/azure-ad/service.js

```diff
@@ -28,11 +28,16 @@ export default Service.extend({
   login() {
     const provider     = get(this, 'access.providers').findBy('id', 'azuread');
     const authRedirect = get(provider, 'redirectUrl');
-    let redirect     = Util.addQueryParams(authRedirect, additionalRedirectParams);
-
-    redirect = Util.addQueryParams(redirect, { state: this.oauth.encodeState(this.oauth.generateState('azuread')) });
+    let redirect = authRedirect;
 
+    Object.keys(additionalRedirectParams).forEach((key) => {
+      if (!authRedirect.includes(key)){
+        redirect = Util.addQueryParam(redirect, key, additionalRedirectParams[key])
+      }
+    })
 
+    redirect = Util.addQueryParams(redirect, { state: this.oauth.encodeState(this.oauth.generateState('azuread')) });
+    redirect = Util.absoluteUrl(redirect)
```

Comment:
Just out of curiosity, why do we do this? How does this transform the URL?

Reply:
We do that because I forgot to remove it :( Thanks for catching that; I was trying it out to see if it'd clean up duplicated keys then forgot to remove it.